Feature/22 pip package

toddlees opened the pull request.

lbeaufort commented on fecfile_validate_python/pyproject.toml :
```diff
@@ -0,0 +1,6 @@
+    "setuptools>=42",
+    "wheel"
+]
+build-backend = "setuptools.build_meta"
```
I noticed there are a few files without newlines at the end - I think there's a VS code setting that will automatically add one for you.

toddlees replied:
done

lbeaufort reviewed (commented):
Just a few small comments from me

lbeaufort commented on fecfile_validate_python/LICENSE:
```diff
@@ -0,0 +1 @@
+This project is covered by the FEC's https://github.com/fecgov/FEC/blob/master/LICENSE.md.
```
Should this be a `.md` file? Also, can the FEC's license be a hyperlink?

toddlees replied:
I based it off of this documentation from PyPa: https://packaging.python.org/en/latest/tutorials/packaging-projects/
I can't find and example of the package license file supporting markdown. pytest and flake8 just have raw text. 
https://github.com/pytest-dev/pytest/blob/main/LICENSE
https://github.com/PyCQA/flake8/blob/main/LICENSE

mjtravers reviewed (changes requested):
A couple of issues:
1. Are there instructions on how to run the pip command to install from the GitHub repo in a README somewhere?
2. The validate() function does not match the API described in the ticket. The validate function is to be more generic and accept the unique schema identifier as a parameter. Currently, the validate() function is hard coded to the F3X schema.

toddlees commented:
@mjtravers, I added the install instructions to the readme.  NOTE: I future proofed it by putting the main branch in the example url.  To get it to work before these commits are in main, you'll need to point to this branch
```pip install "git+https://github.com/fecgov/fecfile-validate@feature/22-pip-package#egg=fecfile_validate&subdirectory=fecfile_validate_python"```

I also made the validate generic and updated the tests.

mjtravers reviewed (changes requested):
Almost there. In the description of the ticket, the return of the validate() function is of this form:

{
'errors': [
{ 'path': 'property_name', 'message': 'message to user describing error' },
...
],
'warnings': []
}

This is to have a spot for the warnings when we add them later and to keep consistent what we're building on the javascript side.

mjtravers reviewed (approved):
Great suite of tests. Thanks for updating the README
